src/components/HeroSection.vue

```diff
@@ -92,15 +94,24 @@ export default {
       setSearchTerm: SET_Q,
       setSearchType: SET_SEARCH_TYPE,
     }),
+    ...mapActions(FILTER, {
+      checkFilter: TOGGLE_FILTER,
+    }),
     getPath() {
       if (!process.env.enableAudio) return '/search/image'
       return `/search/${this.form.searchType}`
     },
     getMediaType() {
       return this.form.searchType
     },
+    toggleFilter({ code, checked }) {
+      this.filters[code] = checked
+    },
     onSubmit() {
       this.setSearchTerm({ q: this.form.searchTerm })
+      Object.keys(this.filters)
+        .filter((filterName) => !!this.filters[filterName])
+        .map((f) => this.checkFilter({ filterType: 'licenseTypes', code: f }))
```

Comment:
This can just be a `forEach` instead of a `map` I think. Might save some tiny amount of CPU cycles by not throwing away the `map` result 😅 Also it's more semantically obvious what is meant... usually with a `map` or `reduce` operation I expect the result to be used, whereas `forEach` is explicitly just a loop over the array and nothing more.

Reply:
Thank you for your explanation :)